src/pages/AppList.svelte

```diff
@@ -12,11 +12,6 @@
 
   const URL = "data/apps.json";
 
-  $pageState = {
-    search: "",
-    ...$pageState,
-  };
-
```

Comment:
Looking at this again I don't think this should be removed, at least the resetting search part. I think it makes sense to keep resetting search state in this component: expected behavior is empty search whenever we click on the homepage right?

Reply:
Not necessarily. If there is a search in the URL (e.g. `?search=android`) we want to bring that into the component. That's the purpose of the `replaceState` function I just added: make sure that *only* the URL parameters find their way into the component.

Reply:
You're right. I was thinking if a user searches for something in the search bar in AppDetail and then clicks on home page, that search state will carry over to AppList which I'm not sure looks right. How should we handle that?

Reply:
In the last two revisions, we synchronize pageState with the URL when navigating here:

https://github.com/mozilla/glean-dictionary/pull/780/files#diff-d68daa8b73cec8e4419759802bbd91e9f87f8ee02f25cdb1d12582ef6de848dbR63

This seems to take care of the problem.